src/vm/memory.zig

```diff
@@ -13,32 +14,74 @@ pub const HEAP_START: u64 = 0x300000000;
 pub const INPUT_START: u64 = 0x400000000;
 const VIRTUAL_ADDRESS_BITS = 32;
 
+pub const AccessError = error{ StackAccessViolation, AccessViolation };
+pub const RegionError = AccessError || error{InvalidMemoryRegion};
+pub const InitError = error{InvalidMemoryRegion};
+
+// [agave] https://github.com/anza-xyz/sbpf/blob/a8247dd30714ef286d26179771724b91b199151b/src/memory_region.rs#L731
 pub const MemoryMap = union(enum) {
     aligned: AlignedMemoryMap,
-    // TODO: unaligned memory map?
+    unaligned: UnalignedMemoryMap,
+
+    pub fn init(
+        allocator: std.mem.Allocator,
+        regions: []const Region,
+        version: sbpf.Version,
+        config: *const exe.Config,
+    ) (error{OutOfMemory} || InitError)!MemoryMap {
+        return if (config.aligned_memory_mapping)
+            .{ .aligned = try AlignedMemoryMap.init(regions, version, config) }
+        else
+            .{ .unaligned = try UnalignedMemoryMap.init(allocator, regions, version, config) };
+    }
 
-    pub fn init(regions: []const Region, version: sbpf.Version) !MemoryMap {
-        return .{ .aligned = try AlignedMemoryMap.init(regions, version) };
+    pub fn deinit(self: *const MemoryMap, allocator: std.mem.Allocator) void {
+        switch (self.*) {
+            .aligned => {},
+            .unaligned => |unaligned| unaligned.deinit(allocator),
+        }
     }
 
-    pub fn region(self: MemoryMap, vm_addr: u64) !Region {
+    fn find_region(
```

Comment:
should be camelCase

Reply:
changed to camelCase in e2a0e3c58e0f908854d59b9212eca28bc5f1bd0e